api/payments.js

```diff
@@ -448,12 +442,15 @@ function getAccountPayments(request, response, next) {
   };
 
   function attachResourceId(transactions, callback) {
-    async.map(transactions, function(payment, async_map_callback) {
+    async.map(transactions, function(paymentResult, async_map_callback) {
+      var hash = paymentResult.hash;
+      var payment = paymentResult.payment;
+
       if (!payment) {
         async_map_callback(new Error('No payment found'));
```

Comment:
We should validate the hash at this point (before passing it into dbinterface.getTransaction())